### DISCUSSION TITLE
[Question][HW3] coverage of instruction without metadata

### DISCUSSION THREAD

**Discussion by AnHaechan** [prosyslab-classroom/cs348-information-security#550] [Q&A]
Should we instrument the `__coverage__` function call 
for the instruction without any `!dbg` metadata?

For example, LOC 23 of example3.ll is 
```
if.else:                                         
%div1 = sdiv i32 40, %a, !dbg !21
call void @llvm.dbg.value(metadata i32 %div1, metadata !14, metadata !DIExpression()), !dbg !12  
br label %if.end ; LOC23
```
It does not matter for the pass/fail of test3 because the control flow never reaches to that instruction,
but I've got curious about general cases.

Should we consider making some hacky way to this case(where instruction does not have `!debug` metadata?)

**Comment by KihongHeo** [comment]
No. Only consider instructions that have debug information.
